core/src/main/java/org/jboss/gwt/elemento/core/BodyObserver.java

```diff
@@ -0,0 +1,149 @@
+package org.jboss.gwt.elemento.core;
+
+import elemental2.dom.*;
+import jsinterop.base.Js;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static elemental2.dom.DomGlobal.document;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+class BodyObserver {
+
+    private static String ATTACH_UID_KEY = "on-attach-uid";
+    private static String DETACH_UID_KEY = "on-detach-uid";
+
+    private static List<ElementObserver> detachObservers = new ArrayList<>();
+    private static List<ElementObserver> attachObservers = new ArrayList<>();
+
+    static {
+        MutationObserver mutationObserver = new MutationObserver((MutationRecord[] records, MutationObserver observer) -> {
+            for (int i = 0; i < records.length; i++) {
+                onElementsRemoved(records[i]);
+                onElementsAppended(records[i]);
+            }
+            return null;
+        });
+
+        MutationObserverInit mutationObserverInit = MutationObserverInit.create();
+        mutationObserverInit.setChildList(true);
+        mutationObserverInit.setSubtree(true);
+
+        mutationObserver.observe(document.body, mutationObserverInit);
+    }
+
```

Comment:
Can you start observing via lazy loading instead of doing it in a static block? 
When the first onAttach/onDetach is called, observing should get started.

Doing it in a static block means that every application using elemento will have a running observer (even if it does not use the functionality).

Reply:
yup, i will make it lazy.

Reply:
I regarding this one i think that unless you use this `BodyObeserver` the code will be trimmed out by the compiler, which means there will be no Observer attached to the body unless you really used the class.

Reply:
but if the class is used but actually the code using it is not yet called it will attach the observer even before that happens.

Reply:
I think this depends how the gwt (or J2CL) compiler handles this. Because BodyObeserver is imported in the Elements class - and everybody will use the Elements class - i'm not 100% sure what happens / will happen in the future. Doing this via lazy loading will be the safe variant...